src/pages/dev/axelarjs-sdk/tx-status-query-recovery.mdx

```diff
@@ -1,18 +1,30 @@
-# Query and recover GMP transactions
+# Programmatically query and recover GMP transactions
 
-Occasionally, transactions can get "stuck" in the pipeline from a source to destination chain (e.g. due to one-off issues that arise with relayers that operate on top of the network).
+import { Callout } from "/src/components/callout"
 
-The `AxelarGMPRecoveryAPI` module in the AxelarJS SDK can be used by your dApp to query the status of any General Message Passing (GMP) transaction (triggered by either `callContract` or `callContractWithToken`) on the gateway contract of a source chain and trigger a manual relay from source to destination if necessary. - The [GMP status tracker](/dev/general-message-passing/debug/transaction-recovery/) on Axelarscan makes use of this feature.
+Transactions can occasionally get stuck in the pipeline from a source to destination chain, mostly from one of the two following reasons:
 
-### Install the AxelarJS SDK module (`AxelarGMPRecoveryAPI`)
+- The transaction is not relayed from the source chain into the Axelar network for processing.
+- The transaction fails to get executed on the destination chain.
+
+The [`AxelarGMPRecoveryAPI`](https://github.com/axelarnetwork/axelarjs-sdk/blob/main/src/libs/TransactionRecoveryApi/AxelarGMPRecoveryAPI.ts) module in the AxelarJS SDK can be used to troubleshoot:
```

Comment:
It has lots of functionality here even beyond querying tx status & manually relaying so maybe we can say it can be helpful in this context and some of the functionality it has are queryTxStatus and manualRelay()

I do think we should try mention more of the functionality that it offers beyond just those two functions though like; `doesTxMeetConfirmHt`, `getEvmEvent`, `isExecuted` etc. Also could be good to link to the actual functions in the source code for people to explore deeper.

Reply:
Sounds good. Let's push these changes now so the docs are no longer broken, and work together tomorrow or next week to properly update this page. I'll set up a meeting for us to sync.